### PR TITLE
Fix response of the resend confirmation code request

### PR DESCRIPTION
**Pull request by lammertw**
@gaebel In PR https://github.com/Liftric/cognito-idp/pull/36 you asked me to use `CodeDeliveryDetails` as response type. However I made a mistake in the change there. Instead of wrapping the `CodeDeliveryDetails` in a new response object it's trying to parse the response json (which contains looks like `{"CodeDeliveryDetails":{"AttributeName":"phone_number","DeliveryMedium":"SMS","Destination":"+********0388"}}`) directly into a CodeDeliveryDetails. Which now results in an error. So this call currently always fails.

I've fixed it in this PR (and would greatly appreciate it if a new release could be made with this change since we currently have the error in our app on production, tnx!)

### CODE REVIEW

**Comment by gaebel** [comment]
Hi @lammertw, I forgot to write a test for this, my bad! I'll merge your PR and publish a new release in the morning. Could you also make the change in the IdentityProviderJS class and write the missing test case?

**Comment by lammertw** [comment]
@gaebel Sure, I've fixed JS as well (basically changed it to match the forgot password response since it's the same). I'd like to add some test, however when I'm running the existing tests none of them work. Not sure if I'm missing anything. I created a new Cognito instance on AWS for this and use that when creating the IdentityProviderClient, however I'm getting this:
```
actual value is not null expected null, but was:<com.liftric.cognito.idp.core.IdentityProviderException$NotAuthorized: A client attempted to write unauthorized attribute>
```

File: src/jsMain/kotlin/IdentityProviderJS.kt [hunk 1]
```diff
@@ -28,10 +28,10 @@ class IdentityProviderClientJS(region: String, clientId: String) {
                 .getOrThrow()
         }
 
-    fun resendConfirmationCode(username: String): Promise<CodeDeliveryDetailsJS> =
+    fun resendConfirmationCode(username: String): Promise<ResendConfirmationCodeResponseJS> =
         MainScope().promise {
-            provider.resendConfirmationCode(username)
-                .getOrThrow().let { it.toJs() }
+            provider.forgotPassword(username)
```
Comment (gaebel) [inline src/jsMain/kotlin/IdentityProviderJS.kt:33]: I'd prefer using `resendConfirmationCode` as the underlying call

File: src/jsMain/kotlin/IdentityProviderJS.kt [hunk 2]
```diff
@@ -28,10 +28,10 @@ class IdentityProviderClientJS(region: String, clientId: String) {
                 .getOrThrow()
         }
 
-    fun resendConfirmationCode(username: String): Promise<CodeDeliveryDetailsJS> =
+    fun resendConfirmationCode(username: String): Promise<ResendConfirmationCodeResponseJS> =
```
Comment (gaebel) [inline src/jsMain/kotlin/IdentityProviderJS.kt:31]: In that case we could directly return `CodeDeliveryDetailsJS`

**Comment by gaebel** [comment]
> @gaebel Sure, I've fixed JS as well (basically changed it to match the forgot password response since it's the same). I'd like to add some test, however when I'm running the existing tests none of them work. Not sure if I'm missing anything. I created a new Cognito instance on AWS for this and use that when creating the IdentityProviderClient, however I'm getting this:
> 
> ```
> actual value is not null expected null, but was:<com.liftric.cognito.idp.core.IdentityProviderException$NotAuthorized: A client attempted to write unauthorized attribute>
> ```

I think writable custom attributes have to be enabled in the user pool.

File: src/jsMain/kotlin/IdentityProviderJS.kt (hunk 2 above)
Comment (lammertw) [reply src/jsMain/kotlin/IdentityProviderJS.kt:31]: Why is that? All other functions return a ResponseJS object as far as I can see.

**Comment by lammertw** [comment]
@gaebel I finally got the tests to work. I assume you haven't documented anywhere how to exactly set up the Cognito User Pool but perhaps it's good to know for the future. So some of the things I had to do:
- When creating the user group make sure user account recovery was set to "Email if available, otherwise SMS". This is to ensure that `email` would not become required since the test suite doesn't use emails.
- Add `target_group` as custom attribute
- Add a Pre Signup Lambda that would automatically confirm the user:
```
exports.handler = (event, context, callback) => {
    event.response.autoConfirmUser = true;
    context.done(null, event);
};
```

Because of this setup, it's not possible to add a test case for the `resendConfirmationCode` that I added. I guess for similar reasons the `confirmSignUp` is not included in the test cases. So hopefully you're ok with that and can now merge the PR.

File: src/jsMain/kotlin/IdentityProviderJS.kt (hunk 2 above)
Comment (gaebel) [reply src/jsMain/kotlin/IdentityProviderJS.kt:31]: Just thought it would be fine since it only stores the `CodeDeliveryDetails`. But yeah, might be better to leave it as is for consistency and to reflect the actual returned AWS object.

**Comment by gaebel** [comment]
Yes, that's unfortunately not documented, thanks for the writeup @lammertw – I'll add it to the Contributing.md. Oh... yeah this can't currently be tested in this repo. We're testing the confirmation requests (except `resendConfirmationCode`) in our private project, though. I may add this missing piece to this repo in the coming weeks.
